dot/state/epoch.go

```diff
@@ -236,6 +254,32 @@ func (s *EpochState) GetEpochData(epoch uint64) (*types.EpochData, error) {
 	return raw.ToEpochData()
 }
 
+// GetEpochDataForHeader retrieves the right epoch data that belongs to the header parameter
+func (s *EpochState) GetEpochDataForHeader(epoch uint64, header *types.Header) (*types.EpochData, error) {
```

Comment:
nevermind, I see that in the babe verifier it first checks the database, then checks this for the in-memory epoch data. maybe it would be simpler to have the babe verifier call one function called `GetEpochData` which first checks the database, then the in-memory map?